File: output/ChatGPT/avatar/Java/C++/atcoder_ABC118_C.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> func(vector<int>& A) {
    vector<int> a = A;
    int min = 0;
    for (int i = 0; i < a.size(); i++) {
        if (a[i] == 0) {
            a.erase(a.begin() + i);
            i--;
        }
        else {
            if (min != 0) {
                a[i] %= min;
                if (a[i] == 1) {
                    cout << 1 << endl;
                    exit(0);
                }
            }
            else {
                min = a[i];
            }
        }
    }
    sort(a.begin(), a.end());
    return a;
}

int main() {
    int n;
    cin >> n;
    vector<int> a;
    if (n == 1) {
        int num;
        cin >> num;
        cout << num << endl;
        return 0;
    }
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        a.push_back(num);
    }
    sort(a.begin(), a.end());
    if (a[a.size() - 1] == 0) {
        cout << 0 << endl;
        return 0;
    }
    int ans = 0;
    while (true) {
        if (a.size() == 1) {
            ans = a[0];
            break;
        }
        a = func(a);
    }
    cout << ans << endl;
    return 0;
}

File: output/ChatGPT/avatar/Java/Python/atcoder_ABC118_C.py
import sys

def func(a):
    min_val = 0
    for i in range(len(a)):
        if a[i] == 0:
            a.pop(i)
            i -= 1
        else:
            if min_val != 0:
                a[i] = a[i] % min_val
                if a[i] == 1:
                    print(1)
                    sys.exit(0)
            else:
                min_val = a[i]
    a.sort()
    return a

n = int(input())
a = []
if n == 1:
    print(int(input()))
    sys.exit(0)
for i in range(n):
    a.append(int(input()))

a.sort()
if a[-1] == 0:
    print(0)
    sys.exit(0)

ans = 0
while True:
    if len(a) == 1:
        ans = a[0]
        break
    a = func(a)

print(ans)